### PR TITLE
fix: Do not create secret versions when using Cloud Build repositores second gen

### PR DESCRIPTION
**Pull request by daniel-cit**
This PR removed the creation of not secrets and secret versions in Terraform when using Cloud Build repositores second gen

additionally this PR also externalized the `auto_create_network` variable of the projects created by the modules

### CODE REVIEW

**Comment by daniel-cit** [comment]
@bharathkkb @apeabody 
Could you PTAL ?

File: modules/cloudbuild/main.tf [hunk 1]
```diff
@@ -108,6 +109,16 @@ resource "time_sleep" "impersonate_propagation" {
   ]
 }
 
+/******************************************
+  Cloudbuild Service Agent Role
+******************************************/
+
+resource "google_project_iam_member" "cb_service_agent_role" {
```
Comment (apeabody) [inline modules/cloudbuild/main.tf:116]: This should be fine, but should be automatically granted: https://cloud.google.com/build/docs/securing-builds/configure-access-for-cloud-build-service-account#service-agent-permissions

I wonder if the propagation time is causing an issue with the CI?

File: test/fixtures/cloudbuild_repo_connection_github/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+variable "project_id" {
+  description = "The ID of the project in which to provision resources."
+  type        = string
+}
+
+variable "github_pat" {
+  description = "The personal access token for authenticating with GitHub."
+  type        = string
```
Comment (apeabody) [inline test/fixtures/cloudbuild_repo_connection_github/variables.tf:24]: Perhaps mark as `sensitive   = true`?

File: test/fixtures/cloudbuild_repo_connection_gitlab/variables.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+variable "repository_name" {
+  description = "The name of the test repository."
+  type        = string
+}
+
+variable "gitlab_authorizer_credential" {
+  description = "Credential for GitLab authorizer"
+  type        = string
```
Comment (apeabody) [inline test/fixtures/cloudbuild_repo_connection_gitlab/variables.tf:34]: Not familiar with this credential, should they be `sensitive   = true`?

File: test/fixtures/tf_cloudbuild_builder_simple_github/variables.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The ID of the project in which to provision resources."
+  type        = string
+}
+
+variable "github_pat" {
+  description = "The personal access token for authenticating with GitHub."
```
Comment (apeabody) [inline test/fixtures/tf_cloudbuild_builder_simple_github/variables.tf:23]: `sensitive   = true`?

File: test/fixtures/tf_cloudbuild_workspace_simple_github/variables.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+variable "project_id" {
+  description = "The ID of the project in which to provision resources."
+  type        = string
+}
+
+variable "github_pat" {
+  description = "The personal access token for authenticating with GitHub."
+  type        = string
```
Comment (apeabody) [inline test/fixtures/tf_cloudbuild_workspace_simple_github/variables.tf:24]: `sensitive   = true`?

**Review by apeabody — COMMENTED** [review]
Thanks @daniel-cit!  A few questions before we merge.

File: test/fixtures/cloudbuild_repo_connection_github/variables.tf (hunk 2 above)
Comment (daniel-cit) [reply test/fixtures/cloudbuild_repo_connection_github/variables.tf:24]: done

File: test/fixtures/cloudbuild_repo_connection_gitlab/variables.tf (hunk 3 above)
Comment (daniel-cit) [reply test/fixtures/cloudbuild_repo_connection_gitlab/variables.tf:34]: done

File: test/fixtures/tf_cloudbuild_builder_simple_github/variables.tf (hunk 4 above)
Comment (daniel-cit) [reply test/fixtures/tf_cloudbuild_builder_simple_github/variables.tf:23]: done

File: test/fixtures/tf_cloudbuild_workspace_simple_github/variables.tf (hunk 5 above)
Comment (daniel-cit) [reply test/fixtures/tf_cloudbuild_workspace_simple_github/variables.tf:24]: done

**Pull request merged**